Add per-component build_timeout for system test Docker builds (#848)
precice/tutorials: tools/tests/README.md

### Timeouts

> Build and run/compare use separate timeouts.

Reviewer: ```suggestion
The build and and the run/compare steps use separate timeouts.
```

> **Build:** Each component in `components.yaml` may define `build_timeout` (seconds). For a test, the build step uses the maximum `build_timeout` among its components; components without `build_timeout` use `DEFAULT_BUILD_TIMEOUT` (600s by default, same as `GLOBAL_TIMEOUT`). Override the default via `PRECICE_SYSTEMTESTS_BUILD_TIMEOUT`.

Reviewer: Does that mean that if I have a test that uses OpenFOAM and DuMux, building the OpenFOAM Docker image will timeout at the DuMux build timeout? Why is that?

Author: docker compose build is a single subprocess for all images , so there is one wall-clock limit , we take the maximum so the slowest adapter is not cut off early ; faster images are not capped to a shorter per-component limit , clarified this in the README and in _resolve_build_timeout s docstring.

> **Build:** Each component in `components.yaml` may define `build_timeout` (seconds). The build step runs `docker compose build` once for all participant images, with one wall-clock subprocess timeout. That limit is the maximum `build_timeout` among the test's distinct components (so the slowest adapter, e.g. DuMuX or SU2, is not cut off early). Components without `build_timeout` contribute `DEFAULT_BUILD_TIMEOUT` (600s by default). Override the default via `PRECICE_SYSTEMTESTS_BUILD_TIMEOUT`.

Reviewer: Plus some minor changes to make this shorter, easier to read, and more accurate.

We generally don't need to refer to the exact variables in the scripts, if some section is user-facing.

```suggestion
**Build:** Each component in `components.yaml` may set a `build_timeout` (default: 600s). The build step runs `docker compose build` once for all participant images, with one wall-clock subprocess timeout. That limit is the maximum `build_timeout` among the distinct components of the test (so that the slowest component is not cut off too early). You can override the default via the `PRECICE_SYSTEMTESTS_BUILD_TIMEOUT` environment variable.
```

> **Run and compare:** `GLOBAL_TIMEOUT` defaults to 300s and can be overridden via `PRECICE_SYSTEMTESTS_TIMEOUT`. Tests can define a different `timeout` in their `tests.yaml` entry, which applies to the running and results comparison steps only.

Reviewer: ```suggestion
**Run and compare:** Each test in `tests.yaml` may set a `timeout` (default: 300s), which applies to the running and results comparison steps only. You can override the default via the `PRECICE_SYSTEMTESTS_TIMEOUT` environment variable.
```

### Timeouts

The build and the run/compare steps use separate timeouts.

**Build:** Each component in `components.yaml` may set a `build_timeout` (default: 600s). The build step runs `docker compose build` once for all participant images, with one wall-clock subprocess timeout. That limit is the maximum `build_timeout` among the distinct components of the test (so that the slowest component is not cut off too early). You can override the default via the `PRECICE_SYSTEMTESTS_BUILD_TIMEOUT` environment variable.

> **Run and compare:** Each test in `tests.yaml` may set a `timeout` (default: 300s), which applies to the running and results comparison steps only. You can override the default via the `PRECICE_SYSTEMTESTS_TIMEOUT` environment variable.

Reviewer: ```suggestion
**Build:** Each component in `components.yaml` may set a `build_timeout` (default: 480s (8min)). The build step runs `docker compose build` once for all participant images, with one wall-clock subprocess timeout. That limit is the maximum `build_timeout` among the distinct components of the test (so that the slowest component is not cut off too early). You can override the default via the `PRECICE_SYSTEMTESTS_BUILD_TIMEOUT` environment variable.

**Run and compare:** Each test in `tests.yaml` may set a `timeout` (default: 180s (3min)), which applies to the running and results comparison steps only. You can override the default via the `PRECICE_SYSTEMTESTS_TIMEOUT` environment variable.
```
